build/templates/session.py.mako

```diff
@@ -97,13 +97,19 @@ You can specify a subset of repeated capabilities using the Python index notatio
 '''
 if attributes[attribute]['channel_based'] == 'True':
     attributes[attribute]['documentation']['tip'] = rep_cap_attr_desc.format(attributes[attribute]["name"].lower())
+
+# Handle attribute names that start with a number
+beginning_underscore = ''
+if attributes[attribute]['name'][0].isdigit():
+    beginning_underscore = '_'
+
 %>\
     %if attributes[attribute]['enum']:
-    ${attributes[attribute]['name'].lower()} = attributes.AttributeEnum(attributes.Attribute${attributes[attribute]['type']}, enums.${attributes[attribute]['enum']}, ${attribute})
+    ${beginning_underscore + attributes[attribute]['name'].lower()} = attributes.AttributeEnum(attributes.Attribute${attributes[attribute]['type']}, enums.${attributes[attribute]['enum']}, ${attribute})
```

Comment:
Is the python_name for the attribute not in metadata?
That seems weirdly inconsistent with function.

Assuming I understand correctly, this should be moved to "metadata expansion" phase.

Reply:
Moved to add_all